### DISCUSSION TITLE
Flowise not working as it should

### DISCUSSION THREAD

**Discussion by amis3258** [community/community#78536] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Hi All
I'm not a coder (anymore, haven't coded in 15 years) so I wanted try a simple no code example. I have have access to LangChain via Flowise, which I downloaded from the GitHub repository and set it up via Render. I have created a chatbot to chat to PDF. I have used drag and drop components in Flowise and connected all the components. The DB is Pinecone, a vector DB. All apps I have used are free versions, as I'm just experimenting. All is working but component that allows you to upload the PDF, doesnt automatically upsert the content of the PDF to Pinecone. I have to click on a button to the top right of my Flowise project to upsert to Pinecone. is this normal? I would have expected me clicking on the upload PDF to upsert to pinecone automatically. The components I have connected together in Flowise are: Recursive Character Text Splitter; PDF File; OpenAI embeddings; ChatOpenAI; Pinecone and Conversational Retrival QA chain.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @amis3258 ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. 

If you're asking how to use the piece of software that you found in a repository on GitHub, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.

**Comment by lancerkind** [comment, marked as answer]
Hi Amise, the upload of the PDFs doesn’t upload to the database. It uploads into the Flowise app so it is ready when the workflow is activated.

**Comment by ravindudil5han** [comment]
In a typical no-code workflow, the execution of actions, such as upserting content to Pinecone, might be triggered by specific events or user interactions. If you find that you need to manually click a button to initiate the upsert process, it could be by design in the Flowise platform or the components you've connected.

In no-code platforms, the flow of actions is often determined by how components are linked or configured. Here are a few things to check:

1. **Component Configuration:**
   Review the configuration settings of the components, especially the PDF File and Pinecone components. Ensure that they are set up to automatically trigger the upsert operation when a PDF is uploaded.

2. **Event Trigger:**
   Check if there is an event trigger associated with the PDF upload. Some no-code platforms allow you to define triggers (like "On Upload") that initiate subsequent actions.

3. **Documentation:**
   Refer to the documentation or user guides of Flowise and the components you're using. There might be specific instructions or features related to automatic upserts that you need to configure.

4. **Support Forums:**
   If you encounter difficulties, consider checking the support forums or community discussions related to Flowise or the components you're using. Others may have experienced similar issues and found solutions.

5. **Contact Support:**
   If the behavior seems unexpected and you've exhausted the available resources, reaching out to the support team of Flowise or Pinecone could provide specific assistance tailored to your setup.

Remember that the behavior of no-code platforms can vary, and the configuration options might differ between different services. If there's no built-in automation for the PDF upload triggering the upsert to Pinecone, you might need to manually add a step or configuration to achieve the desired automation in your workflow.
